### PR TITLE
Prepare CHANGELOG/VERSION for 0.58.0 release

### PR DESCRIPTION
**Pull request by monkut**
✏️ update version 0.57.0 -> 0.58.0
✏️ update CHANGELOG.md with issues addressed in 0.58.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Zappa Changelog
 
+## 0.58.0
+
+* support new ephemeral storage feature in zappa_settings.json (#1120)
+* Update permissions (PR #1119)
+* Outdated manylinux wheels download logic (#1249)
+* cryptography>=35.0, plus pip>=20.3 - downloads wrong cryptography anywheel package (GLIBC_2.18 error) (#1063)
+* fix response time improperly configured for micro-seconds. (#1265)
+* fix unquote issue with querystring handling. (#1264)
```
Comment (grantmcconnaughey) [inline CHANGELOG.md:10]: Plus Python 3.11 support. 😄

Comment (monkut) [reply CHANGELOG.md]: oops...

**Review by hellno — APPROVED** [review]

**Review by goya813 — APPROVED** [review]

**Review by souravjamwal77 — APPROVED** [review]
I have already tested the ephemeral storage, python 3.11 and, outdated manylinux wheels PRs. 
Approving for the release.
I will further test each feature once we get a release.

**Comment by hayd** [comment]
This was released to pypi on the 15th! 🥳 
Would it be ok to release/tag this in github too? Thanks!

**Pull request merged**